File: app/front/src/components/main/dto/wktbl_history/updateWkTblHistoryPersonResultDto.ts
import { WkTblKanrenshaPersonHistoryEntity, type WkTblKanrenshaPersonHistoryEntityInterface } from "../../entity/wkTblKanrenshaPersonHistoryEntity";
import { FrameworkMessageAndResultDto, type FrameworkMessageAndResultDtoInterface } from "seijishikin-jp-normalize_common-tool";

interface UpdateWkTblHistoryPersonResultDtoInterface extends FrameworkMessageAndResultDtoInterface {

    /** 編集対象Entity */
    wkTblKanrenshaPersonHistoryEntity: WkTblKanrenshaPersonHistoryEntityInterface;

}

/**
 * 個人一括登録履歴ワークテーブル更新CapsuleDto
 */
class UpdateWkTblHistoryPersonResultDto extends FrameworkMessageAndResultDto
    implements UpdateWkTblHistoryPersonResultDtoInterface {

    /** 編集対象Entity */
    wkTblKanrenshaPersonHistoryEntity: WkTblKanrenshaPersonHistoryEntityInterface;

    constructor() {
        super();
        this.wkTblKanrenshaPersonHistoryEntity = new WkTblKanrenshaPersonHistoryEntity();
    }

}

export { type UpdateWkTblHistoryPersonResultDtoInterface, UpdateWkTblHistoryPersonResultDto }

File: app/back/src/main/java/net/seijishikin/jp/normalize/manage/kanrensha/dto/wktbl_history/UpdateWkTblHistoryPersonResultDto.java
package net.seijishikin.jp.normalize.manage.kanrensha.dto.wktbl_history;

import java.io.Serializable;

import net.seijishikin.jp.normalize.common_tool.dto.FrameworkMessageAndResultDto;
import net.seijishikin.jp.normalize.manage.kanrensha.entity.WkTblKanrenshaPersonHistoryEntity;

/**
 * 個人一括登録履歴ワークテーブル更新CapsuleDto
 */
public class UpdateWkTblHistoryPersonResultDto extends FrameworkMessageAndResultDto // NOPMD DataClass
        implements Serializable {

    /** Serialize id */
    private static final long serialVersionUID = 1L;

    /** 編集対象Entity */
    private WkTblKanrenshaPersonHistoryEntity wkTblKanrenshaPersonHistoryEntity = new WkTblKanrenshaPersonHistoryEntity();

    /**
     * 編集対象Entityを取得する
     * 
     * @return 編集対象Entity
     */
    public WkTblKanrenshaPersonHistoryEntity getWkTblKanrenshaPersonHistoryEntity() {
        return wkTblKanrenshaPersonHistoryEntity;
    }

    /**
     * 編集対象Entityを設定する
     * 
     * @param wkTblKanrenshaPersonHistoryEntity 編集対象Entity
     */
    public void setWkTblKanrenshaPersonHistoryEntity(
            final WkTblKanrenshaPersonHistoryEntity wkTblKanrenshaPersonHistoryEntity) {
        this.wkTblKanrenshaPersonHistoryEntity = wkTblKanrenshaPersonHistoryEntity;
    }

}
